File: micronaut/src/main/java/io/swagger/model/V3Outlet.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.validation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * V3Outlet
 */
@Validated
@Introspected
@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.MicronautCodegen", date = "2024-04-14T03:36:01.894111500+10:00[Australia/Sydney]")
public class V3Outlet   {
  @JsonProperty("outlet_slid_spid")
  private String outletSlidSpid = null;

  @JsonProperty("outlet_name")
  private String outletName = null;

  @JsonProperty("outlet_business")
  private String outletBusiness = null;

  @JsonProperty("outlet_latitude")
  private Float outletLatitude = null;

  @JsonProperty("outlet_longitude")
  private Float outletLongitude = null;

  @JsonProperty("outlet_suburb")
  private String outletSuburb = null;

  @JsonProperty("outlet_postcode")
  private Integer outletPostcode = null;

  @JsonProperty("outlet_business_hour_mon")
  private String outletBusinessHourMon = null;

  @JsonProperty("outlet_business_hour_tue")
  private String outletBusinessHourTue = null;

  @JsonProperty("outlet_business_hour_wed")
  private String outletBusinessHourWed = null;

  @JsonProperty("outlet_business_hour_thur")
  private String outletBusinessHourThur = null;

  @JsonProperty("outlet_business_hour_fri")
  private String outletBusinessHourFri = null;

  @JsonProperty("outlet_business_hour_sat")
  private String outletBusinessHourSat = null;

  @JsonProperty("outlet_business_hour_sun")
  private String outletBusinessHourSun = null;

  @JsonProperty("outlet_notes")
  private String outletNotes = null;

  public V3Outlet outletSlidSpid(String outletSlidSpid) {
    this.outletSlidSpid = outletSlidSpid;
    return this;
  }

  /**
   * The SLID / SPID
   * @return outletSlidSpid
  **/
  @Schema(description = "The SLID / SPID")

  public String getOutletSlidSpid() {
    return outletSlidSpid;
  }

  public void setOutletSlidSpid(String outletSlidSpid) {
    this.outletSlidSpid = outletSlidSpid;
  }

  public V3Outlet outletName(String outletName) {
    this.outletName = outletName;
    return this;
  }

  /**
   * The location name of the outlet
   * @return outletName
  **/
  @Schema(description = "The location name of the outlet")

  public String getOutletName() {
    return outletName;
  }

  public void setOutletName(String outletName) {
    this.outletName = outletName;
  }

  public V3Outlet outletBusiness(String outletBusiness) {
    this.outletBusiness = outletBusiness;
    return this;
  }

  /**
   * The business name of the outlet
   * @return outletBusiness
  **/
  @Schema(description = "The business name of the outlet")

  public String getOutletBusiness() {
    return outletBusiness;
  }

  public void setOutletBusiness(String outletBusiness) {
    this.outletBusiness = outletBusiness;
  }

  public V3Outlet outletLatitude(Float outletLatitude) {
    this.outletLatitude = outletLatitude;
    return this;
  }

  /**
   * Geographic coordinate of latitude at outlet
   * @return outletLatitude
  **/
  @Schema(description = "Geographic coordinate of latitude at outlet")

  public Float getOutletLatitude() {
    return outletLatitude;
  }

  public void setOutletLatitude(Float outletLatitude) {
    this.outletLatitude = outletLatitude;
  }

  public V3Outlet outletLongitude(Float outletLongitude) {
    this.outletLongitude = outletLongitude;
    return this;
  }

  /**
   * Geographic coordinate of longitude at outlet
   * @return outletLongitude
  **/
  @Schema(description = "Geographic coordinate of longitude at outlet")

  public Float getOutletLongitude() {
    return outletLongitude;
  }

  public void setOutletLongitude(Float outletLongitude) {
    this.outletLongitude = outletLongitude;
  }

  public V3Outlet outletSuburb(String outletSuburb) {
    this.outletSuburb = outletSuburb;
    return this;
  }

  /**
   * The city/municipality the outlet is in
   * @return outletSuburb
  **/
  @Schema(description = "The city/municipality the outlet is in")

  public String getOutletSuburb() {
    return outletSuburb;
  }

  public void setOutletSuburb(String outletSuburb) {
    this.outletSuburb = outletSuburb;
  }

  public V3Outlet outletPostcode(Integer outletPostcode) {
    this.outletPostcode = outletPostcode;
    return this;
  }

  /**
   * The postcode for the outlet
   * @return outletPostcode
  **/
  @Schema(description = "The postcode for the outlet")

  public Integer getOutletPostcode() {
    return outletPostcode;
  }

  public void setOutletPostcode(Integer outletPostcode) {
    this.outletPostcode = outletPostcode;
  }

  public V3Outlet outletBusinessHourMon(String outletBusinessHourMon) {
    this.outletBusinessHourMon = outletBusinessHourMon;
    return this;
  }

  /**
   * The business hours on Monday
   * @return outletBusinessHourMon
  **/
  @Schema(description = "The business hours on Monday")

  public String getOutletBusinessHourMon() {
    return outletBusinessHourMon;
  }

  public void setOutletBusinessHourMon(String outletBusinessHourMon) {
    this.outletBusinessHourMon = outletBusinessHourMon;
  }

  public V3Outlet outletBusinessHourTue(String outletBusinessHourTue) {
    this.outletBusinessHourTue = outletBusinessHourTue;
    return this;
  }

  /**
   * The business hours on Tuesday
   * @return outletBusinessHourTue
  **/
  @Schema(description = "The business hours on Tuesday")

  public String getOutletBusinessHourTue() {
    return outletBusinessHourTue;
  }

  public void setOutletBusinessHourTue(String outletBusinessHourTue) {
    this.outletBusinessHourTue = outletBusinessHourTue;
  }

  public V3Outlet outletBusinessHourWed(String outletBusinessHourWed) {
    this.outletBusinessHourWed = outletBusinessHourWed;
    return this;
  }

  /**
   * The business hours on Wednesday
   * @return outletBusinessHourWed
  **/
  @Schema(description = "The business hours on Wednesday")

  public String getOutletBusinessHourWed() {
    return outletBusinessHourWed;
  }

  public void setOutletBusinessHourWed(String outletBusinessHourWed) {
    this.outletBusinessHourWed = outletBusinessHourWed;
  }

  public V3Outlet outletBusinessHourThur(String outletBusinessHourThur) {
    this.outletBusinessHourThur = outletBusinessHourThur;
    return this;
  }

  /**
   * The business hours on Thursday
   * @return outletBusinessHourThur
  **/
  @Schema(description = "The business hours on Thursday")

  public String getOutletBusinessHourThur() {
    return outletBusinessHourThur;
  }

  public void setOutletBusinessHourThur(String outletBusinessHourThur) {
    this.outletBusinessHourThur = outletBusinessHourThur;
  }

  public V3Outlet outletBusinessHourFri(String outletBusinessHourFri) {
    this.outletBusinessHourFri = outletBusinessHourFri;
    return this;
  }

  /**
   * The business hours on Friday
   * @return outletBusinessHourFri
  **/
  @Schema(description = "The business hours on Friday")

  public String getOutletBusinessHourFri() {
    return outletBusinessHourFri;
  }

  public void setOutletBusinessHourFri(String outletBusinessHourFri) {
    this.outletBusinessHourFri = outletBusinessHourFri;
  }

  public V3Outlet outletBusinessHourSat(String outletBusinessHourSat) {
    this.outletBusinessHourSat = outletBusinessHourSat;
    return this;
  }

  /**
   * The business hours on Saturday
   * @return outletBusinessHourSat
  **/
  @Schema(description = "The business hours on Saturday")

  public String getOutletBusinessHourSat() {
    return outletBusinessHourSat;
  }

  public void setOutletBusinessHourSat(String outletBusinessHourSat) {
    this.outletBusinessHourSat = outletBusinessHourSat;
  }

  public V3Outlet outletBusinessHourSun(String outletBusinessHourSun) {
    this.outletBusinessHourSun = outletBusinessHourSun;
    return this;
  }

  /**
   * The business hours on Sunday
   * @return outletBusinessHourSun
  **/
  @Schema(description = "The business hours on Sunday")

  public String getOutletBusinessHourSun() {
    return outletBusinessHourSun;
  }

  public void setOutletBusinessHourSun(String outletBusinessHourSun) {
    this.outletBusinessHourSun = outletBusinessHourSun;
  }

  public V3Outlet outletNotes(String outletNotes) {
    this.outletNotes = outletNotes;
    return this;
  }

  /**
   * Any additional notes for the outlet such as 'Buy pre-loaded myki cards only'. May be null/empty.
   * @return outletNotes
  **/
  @Schema(description = "Any additional notes for the outlet such as 'Buy pre-loaded myki cards only'. May be null/empty.")

  public String getOutletNotes() {
    return outletNotes;
  }

  public void setOutletNotes(String outletNotes) {
    this.outletNotes = outletNotes;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3Outlet v3Outlet = (V3Outlet) o;
    return Objects.equals(this.outletSlidSpid, v3Outlet.outletSlidSpid) &&
        Objects.equals(this.outletName, v3Outlet.outletName) &&
        Objects.equals(this.outletBusiness, v3Outlet.outletBusiness) &&
        Objects.equals(this.outletLatitude, v3Outlet.outletLatitude) &&
        Objects.equals(this.outletLongitude, v3Outlet.outletLongitude) &&
        Objects.equals(this.outletSuburb, v3Outlet.outletSuburb) &&
        Objects.equals(this.outletPostcode, v3Outlet.outletPostcode) &&
        Objects.equals(this.outletBusinessHourMon, v3Outlet.outletBusinessHourMon) &&
        Objects.equals(this.outletBusinessHourTue, v3Outlet.outletBusinessHourTue) &&
        Objects.equals(this.outletBusinessHourWed, v3Outlet.outletBusinessHourWed) &&
        Objects.equals(this.outletBusinessHourThur, v3Outlet.outletBusinessHourThur) &&
        Objects.equals(this.outletBusinessHourFri, v3Outlet.outletBusinessHourFri) &&
        Objects.equals(this.outletBusinessHourSat, v3Outlet.outletBusinessHourSat) &&
        Objects.equals(this.outletBusinessHourSun, v3Outlet.outletBusinessHourSun) &&
        Objects.equals(this.outletNotes, v3Outlet.outletNotes);
  }

  @Override
  public int hashCode() {
    return Objects.hash(outletSlidSpid, outletName, outletBusiness, outletLatitude, outletLongitude, outletSuburb, outletPostcode, outletBusinessHourMon, outletBusinessHourTue, outletBusinessHourWed, outletBusinessHourThur, outletBusinessHourFri, outletBusinessHourSat, outletBusinessHourSun, outletNotes);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3Outlet {\n");
    
    sb.append("    outletSlidSpid: ").append(toIndentedString(outletSlidSpid)).append("\n");
    sb.append("    outletName: ").append(toIndentedString(outletName)).append("\n");
    sb.append("    outletBusiness: ").append(toIndentedString(outletBusiness)).append("\n");
    sb.append("    outletLatitude: ").append(toIndentedString(outletLatitude)).append("\n");
    sb.append("    outletLongitude: ").append(toIndentedString(outletLongitude)).append("\n");
    sb.append("    outletSuburb: ").append(toIndentedString(outletSuburb)).append("\n");
    sb.append("    outletPostcode: ").append(toIndentedString(outletPostcode)).append("\n");
    sb.append("    outletBusinessHourMon: ").append(toIndentedString(outletBusinessHourMon)).append("\n");
    sb.append("    outletBusinessHourTue: ").append(toIndentedString(outletBusinessHourTue)).append("\n");
    sb.append("    outletBusinessHourWed: ").append(toIndentedString(outletBusinessHourWed)).append("\n");
    sb.append("    outletBusinessHourThur: ").append(toIndentedString(outletBusinessHourThur)).append("\n");
    sb.append("    outletBusinessHourFri: ").append(toIndentedString(outletBusinessHourFri)).append("\n");
    sb.append("    outletBusinessHourSat: ").append(toIndentedString(outletBusinessHourSat)).append("\n");
    sb.append("    outletBusinessHourSun: ").append(toIndentedString(outletBusinessHourSun)).append("\n");
    sb.append("    outletNotes: ").append(toIndentedString(outletNotes)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: swagger-2/csharp/src/IO.Swagger/Model/V3Outlet.cs
/* 
 * PTV Timetable API - Version 3
 *
 * The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.  
 *
 * OpenAPI spec version: v3
 * 
 * Generated by: https://github.com/swagger-api/swagger-codegen.git
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.ComponentModel.DataAnnotations;
using SwaggerDateConverter = IO.Swagger.Client.SwaggerDateConverter;

namespace IO.Swagger.Model
{
    /// <summary>
    /// V3Outlet
    /// </summary>
    [DataContract]
    public partial class V3Outlet :  IEquatable<V3Outlet>, IValidatableObject
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="V3Outlet" /> class.
        /// </summary>
        /// <param name="outletSlidSpid">The SLID / SPID.</param>
        /// <param name="outletName">The location name of the outlet.</param>
        /// <param name="outletBusiness">The business name of the outlet.</param>
        /// <param name="outletLatitude">Geographic coordinate of latitude at outlet.</param>
        /// <param name="outletLongitude">Geographic coordinate of longitude at outlet.</param>
        /// <param name="outletSuburb">The city/municipality the outlet is in.</param>
        /// <param name="outletPostcode">The postcode for the outlet.</param>
        /// <param name="outletBusinessHourMon">The business hours on Monday.</param>
        /// <param name="outletBusinessHourTue">The business hours on Tuesday.</param>
        /// <param name="outletBusinessHourWed">The business hours on Wednesday.</param>
        /// <param name="outletBusinessHourThur">The business hours on Thursday.</param>
        /// <param name="outletBusinessHourFri">The business hours on Friday.</param>
        /// <param name="outletBusinessHourSat">The business hours on Saturday.</param>
        /// <param name="outletBusinessHourSun">The business hours on Sunday.</param>
        /// <param name="outletNotes">Any additional notes for the outlet such as &#39;Buy pre-loaded myki cards only&#39;. May be null/empty..</param>
        public V3Outlet(string outletSlidSpid = default(string), string outletName = default(string), string outletBusiness = default(string), float? outletLatitude = default(float?), float? outletLongitude = default(float?), string outletSuburb = default(string), int? outletPostcode = default(int?), string outletBusinessHourMon = default(string), string outletBusinessHourTue = default(string), string outletBusinessHourWed = default(string), string outletBusinessHourThur = default(string), string outletBusinessHourFri = default(string), string outletBusinessHourSat = default(string), string outletBusinessHourSun = default(string), string outletNotes = default(string))
        {
            this.OutletSlidSpid = outletSlidSpid;
            this.OutletName = outletName;
            this.OutletBusiness = outletBusiness;
            this.OutletLatitude = outletLatitude;
            this.OutletLongitude = outletLongitude;
            this.OutletSuburb = outletSuburb;
            this.OutletPostcode = outletPostcode;
            this.OutletBusinessHourMon = outletBusinessHourMon;
            this.OutletBusinessHourTue = outletBusinessHourTue;
            this.OutletBusinessHourWed = outletBusinessHourWed;
            this.OutletBusinessHourThur = outletBusinessHourThur;
            this.OutletBusinessHourFri = outletBusinessHourFri;
            this.OutletBusinessHourSat = outletBusinessHourSat;
            this.OutletBusinessHourSun = outletBusinessHourSun;
            this.OutletNotes = outletNotes;
        }
        
        /// <summary>
        /// The SLID / SPID
        /// </summary>
        /// <value>The SLID / SPID</value>
        [DataMember(Name="outlet_slid_spid", EmitDefaultValue=false)]
        public string OutletSlidSpid { get; set; }

        /// <summary>
        /// The location name of the outlet
        /// </summary>
        /// <value>The location name of the outlet</value>
        [DataMember(Name="outlet_name", EmitDefaultValue=false)]
        public string OutletName { get; set; }

        /// <summary>
        /// The business name of the outlet
        /// </summary>
        /// <value>The business name of the outlet</value>
        [DataMember(Name="outlet_business", EmitDefaultValue=false)]
        public string OutletBusiness { get; set; }

        /// <summary>
        /// Geographic coordinate of latitude at outlet
        /// </summary>
        /// <value>Geographic coordinate of latitude at outlet</value>
        [DataMember(Name="outlet_latitude", EmitDefaultValue=false)]
        public float? OutletLatitude { get; set; }

        /// <summary>
        /// Geographic coordinate of longitude at outlet
        /// </summary>
        /// <value>Geographic coordinate of longitude at outlet</value>
        [DataMember(Name="outlet_longitude", EmitDefaultValue=false)]
        public float? OutletLongitude { get; set; }

        /// <summary>
        /// The city/municipality the outlet is in
        /// </summary>
        /// <value>The city/municipality the outlet is in</value>
        [DataMember(Name="outlet_suburb", EmitDefaultValue=false)]
        public string OutletSuburb { get; set; }

        /// <summary>
        /// The postcode for the outlet
        /// </summary>
        /// <value>The postcode for the outlet</value>
        [DataMember(Name="outlet_postcode", EmitDefaultValue=false)]
        public int? OutletPostcode { get; set; }

        /// <summary>
        /// The business hours on Monday
        /// </summary>
        /// <value>The business hours on Monday</value>
        [DataMember(Name="outlet_business_hour_mon", EmitDefaultValue=false)]
        public string OutletBusinessHourMon { get; set; }

        /// <summary>
        /// The business hours on Tuesday
        /// </summary>
        /// <value>The business hours on Tuesday</value>
        [DataMember(Name="outlet_business_hour_tue", EmitDefaultValue=false)]
        public string OutletBusinessHourTue { get; set; }

        /// <summary>
        /// The business hours on Wednesday
        /// </summary>
        /// <value>The business hours on Wednesday</value>
        [DataMember(Name="outlet_business_hour_wed", EmitDefaultValue=false)]
        public string OutletBusinessHourWed { get; set; }

        /// <summary>
        /// The business hours on Thursday
        /// </summary>
        /// <value>The business hours on Thursday</value>
        [DataMember(Name="outlet_business_hour_thur", EmitDefaultValue=false)]
        public string OutletBusinessHourThur { get; set; }

        /// <summary>
        /// The business hours on Friday
        /// </summary>
        /// <value>The business hours on Friday</value>
        [DataMember(Name="outlet_business_hour_fri", EmitDefaultValue=false)]
        public string OutletBusinessHourFri { get; set; }

        /// <summary>
        /// The business hours on Saturday
        /// </summary>
        /// <value>The business hours on Saturday</value>
        [DataMember(Name="outlet_business_hour_sat", EmitDefaultValue=false)]
        public string OutletBusinessHourSat { get; set; }

        /// <summary>
        /// The business hours on Sunday
        /// </summary>
        /// <value>The business hours on Sunday</value>
        [DataMember(Name="outlet_business_hour_sun", EmitDefaultValue=false)]
        public string OutletBusinessHourSun { get; set; }

        /// <summary>
        /// Any additional notes for the outlet such as &#39;Buy pre-loaded myki cards only&#39;. May be null/empty.
        /// </summary>
        /// <value>Any additional notes for the outlet such as &#39;Buy pre-loaded myki cards only&#39;. May be null/empty.</value>
        [DataMember(Name="outlet_notes", EmitDefaultValue=false)]
        public string OutletNotes { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class V3Outlet {\n");
            sb.Append("  OutletSlidSpid: ").Append(OutletSlidSpid).Append("\n");
            sb.Append("  OutletName: ").Append(OutletName).Append("\n");
            sb.Append("  OutletBusiness: ").Append(OutletBusiness).Append("\n");
            sb.Append("  OutletLatitude: ").Append(OutletLatitude).Append("\n");
            sb.Append("  OutletLongitude: ").Append(OutletLongitude).Append("\n");
            sb.Append("  OutletSuburb: ").Append(OutletSuburb).Append("\n");
            sb.Append("  OutletPostcode: ").Append(OutletPostcode).Append("\n");
            sb.Append("  OutletBusinessHourMon: ").Append(OutletBusinessHourMon).Append("\n");
            sb.Append("  OutletBusinessHourTue: ").Append(OutletBusinessHourTue).Append("\n");
            sb.Append("  OutletBusinessHourWed: ").Append(OutletBusinessHourWed).Append("\n");
            sb.Append("  OutletBusinessHourThur: ").Append(OutletBusinessHourThur).Append("\n");
            sb.Append("  OutletBusinessHourFri: ").Append(OutletBusinessHourFri).Append("\n");
            sb.Append("  OutletBusinessHourSat: ").Append(OutletBusinessHourSat).Append("\n");
            sb.Append("  OutletBusinessHourSun: ").Append(OutletBusinessHourSun).Append("\n");
            sb.Append("  OutletNotes: ").Append(OutletNotes).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as V3Outlet);
        }

        /// <summary>
        /// Returns true if V3Outlet instances are equal
        /// </summary>
        /// <param name="input">Instance of V3Outlet to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(V3Outlet input)
        {
            if (input == null)
                return false;

            return 
                (
                    this.OutletSlidSpid == input.OutletSlidSpid ||
                    (this.OutletSlidSpid != null &&
                    this.OutletSlidSpid.Equals(input.OutletSlidSpid))
                ) && 
                (
                    this.OutletName == input.OutletName ||
                    (this.OutletName != null &&
                    this.OutletName.Equals(input.OutletName))
                ) && 
                (
                    this.OutletBusiness == input.OutletBusiness ||
                    (this.OutletBusiness != null &&
                    this.OutletBusiness.Equals(input.OutletBusiness))
                ) && 
                (
                    this.OutletLatitude == input.OutletLatitude ||
                    (this.OutletLatitude != null &&
                    this.OutletLatitude.Equals(input.OutletLatitude))
                ) && 
                (
                    this.OutletLongitude == input.OutletLongitude ||
                    (this.OutletLongitude != null &&
                    this.OutletLongitude.Equals(input.OutletLongitude))
                ) && 
                (
                    this.OutletSuburb == input.OutletSuburb ||
                    (this.OutletSuburb != null &&
                    this.OutletSuburb.Equals(input.OutletSuburb))
                ) && 
                (
                    this.OutletPostcode == input.OutletPostcode ||
                    (this.OutletPostcode != null &&
                    this.OutletPostcode.Equals(input.OutletPostcode))
                ) && 
                (
                    this.OutletBusinessHourMon == input.OutletBusinessHourMon ||
                    (this.OutletBusinessHourMon != null &&
                    this.OutletBusinessHourMon.Equals(input.OutletBusinessHourMon))
                ) && 
                (
                    this.OutletBusinessHourTue == input.OutletBusinessHourTue ||
                    (this.OutletBusinessHourTue != null &&
                    this.OutletBusinessHourTue.Equals(input.OutletBusinessHourTue))
                ) && 
                (
                    this.OutletBusinessHourWed == input.OutletBusinessHourWed ||
                    (this.OutletBusinessHourWed != null &&
                    this.OutletBusinessHourWed.Equals(input.OutletBusinessHourWed))
                ) && 
                (
                    this.OutletBusinessHourThur == input.OutletBusinessHourThur ||
                    (this.OutletBusinessHourThur != null &&
                    this.OutletBusinessHourThur.Equals(input.OutletBusinessHourThur))
                ) && 
                (
                    this.OutletBusinessHourFri == input.OutletBusinessHourFri ||
                    (this.OutletBusinessHourFri != null &&
                    this.OutletBusinessHourFri.Equals(input.OutletBusinessHourFri))
                ) && 
                (
                    this.OutletBusinessHourSat == input.OutletBusinessHourSat ||
                    (this.OutletBusinessHourSat != null &&
                    this.OutletBusinessHourSat.Equals(input.OutletBusinessHourSat))
                ) && 
                (
                    this.OutletBusinessHourSun == input.OutletBusinessHourSun ||
                    (this.OutletBusinessHourSun != null &&
                    this.OutletBusinessHourSun.Equals(input.OutletBusinessHourSun))
                ) && 
                (
                    this.OutletNotes == input.OutletNotes ||
                    (this.OutletNotes != null &&
                    this.OutletNotes.Equals(input.OutletNotes))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                if (this.OutletSlidSpid != null)
                    hashCode = hashCode * 59 + this.OutletSlidSpid.GetHashCode();
                if (this.OutletName != null)
                    hashCode = hashCode * 59 + this.OutletName.GetHashCode();
                if (this.OutletBusiness != null)
                    hashCode = hashCode * 59 + this.OutletBusiness.GetHashCode();
                if (this.OutletLatitude != null)
                    hashCode = hashCode * 59 + this.OutletLatitude.GetHashCode();
                if (this.OutletLongitude != null)
                    hashCode = hashCode * 59 + this.OutletLongitude.GetHashCode();
                if (this.OutletSuburb != null)
                    hashCode = hashCode * 59 + this.OutletSuburb.GetHashCode();
                if (this.OutletPostcode != null)
                    hashCode = hashCode * 59 + this.OutletPostcode.GetHashCode();
                if (this.OutletBusinessHourMon != null)
                    hashCode = hashCode * 59 + this.OutletBusinessHourMon.GetHashCode();
                if (this.OutletBusinessHourTue != null)
                    hashCode = hashCode * 59 + this.OutletBusinessHourTue.GetHashCode();
                if (this.OutletBusinessHourWed != null)
                    hashCode = hashCode * 59 + this.OutletBusinessHourWed.GetHashCode();
                if (this.OutletBusinessHourThur != null)
                    hashCode = hashCode * 59 + this.OutletBusinessHourThur.GetHashCode();
                if (this.OutletBusinessHourFri != null)
                    hashCode = hashCode * 59 + this.OutletBusinessHourFri.GetHashCode();
                if (this.OutletBusinessHourSat != null)
                    hashCode = hashCode * 59 + this.OutletBusinessHourSat.GetHashCode();
                if (this.OutletBusinessHourSun != null)
                    hashCode = hashCode * 59 + this.OutletBusinessHourSun.GetHashCode();
                if (this.OutletNotes != null)
                    hashCode = hashCode * 59 + this.OutletNotes.GetHashCode();
                return hashCode;
            }
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}
